src/main/java/ru/bclib/mixin/common/TagLoaderMixin.java

```diff
@@ -1,26 +1,27 @@
 package ru.bclib.mixin.common;
 
 import java.util.Map;
-import java.util.concurrent.Executor;
-import java.util.function.Supplier;
 
+import net.minecraft.server.packs.resources.ResourceManager;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.ModifyArg;
+import org.spongepowered.asm.mixin.injection.Inject;
 
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.Tag;
 import net.minecraft.tags.TagLoader;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 import ru.bclib.util.TagHelper;
 
 @Mixin(TagLoader.class)
 public class TagLoaderMixin {
 	@Shadow
-	private String name;
+	private String directory;
 
-	@ModifyArg(method = "prepare", at = @At(value = "INVOKE", target = "Ljava/util/concurrent/CompletableFuture;supplyAsync(Ljava/util/function/Supplier;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;"))
-	public Supplier<Map<ResourceLocation, Tag.Builder>> be_modifyTags(Supplier<Map<ResourceLocation, Tag.Builder>> supplier, Executor executor) {
-		return () -> TagHelper.apply(name, supplier.get());
+	//@ModifyArg(method = "prepare", at = @At(value = "INVOKE", target = "Ljava/util/concurrent/CompletableFuture;supplyAsync(Ljava/util/function/Supplier;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;"))

```

Comment:
This will break most mods compatibility, there should be ModifyArg instead of Inject

Reply:
Ah, I see. 
As far as I can seen, that call to `prepare` returning the result of `supplyAsync` was removed in 1.17. `load` returns that map without passing it as an argument to another method before it gets returned. Would a TAIL-Inject be ok?

Reply:
No, RETURN and TAIL injects are applied once for mixins (and this depends on mod loading order and some other things), that's why RETURN and TAIL should be used only when interrupting/aborting process is not critical. If the original ModifyArg point doesn't exist anymore - there can be another one in this function